students/Mintae_K/lesson04/mailroom2.py

```diff
@@ -0,0 +1,69 @@
+# Sending a Thank you
+def sendingthank(donors):
+    f_name = input('Please type the Full Name or list> ')
+    while f_name == 'list':
+        print(' ')
+        for x in donors:
+            print(x)
+        print(' ')
+        f_name = input('Please type the Full Name or list> ')
+    if donors.get(f_name, False):
+        s = 'Existing donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors[f_name].append(float(amount))
+    else:
+        s = 'New donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors[f_name] = [float(amount)]
+    print('\n', "Thank you {} for the generous donation!".format(f_name), '\n')
+    return
+
+
+# Creating Report
+def createreport(donors):
+    s = '\n, {:<20}| {:<10} | {:<10}| {:<20}'
+    print(s.format('Donor Name', 'Total Given', "Num Gift", "Average Gift"))
+    print('-' * 60)
+    total = []
+    for keys in donors:
```

Comment:
it is preferred to use `items()` when you need both keys and values; why? as you are creating variables for key and value in a loop, you are assigning good variable names which help the reader.

for example
```
for donor, donations in donors.items()
```
or 
```
for name, donations in donors.items()
```
are clearer than `donors[key]`